File: src/main/java/ru/flestiz/ecosmetics/events/PlayerDataLoadEvent.java
package ru.flestiz.ecosmetics.events;

import ru.flestiz.ecosmetics.api.Cosmetic;
import ru.flestiz.ecosmetics.cache.PlayerData;
import ru.flestiz.ecosmetics.utils.MCNotInstalledException;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

import java.util.Set;

/**
 *  The event is called when the player's data has just been loaded.
 *  This event is asynchronous
 */
public class PlayerDataLoadEvent extends Event {

    public PlayerDataLoadEvent(PlayerData playerData, Set<Cosmetic> equippedCosmetics){
        super(true);
        throw new MCNotInstalledException();
    }

    @Override
    public HandlerList getHandlers() {
        throw  new MCNotInstalledException();
    }

    public static HandlerList getHandlerList() {
        throw  new MCNotInstalledException();
    }

    /**
     * Player whose data has been uploaded
     * @return PlayerData
     */
    public PlayerData getPlayerData() {
        throw  new MCNotInstalledException();
    }

    /**
     * Cosmetics that have been loaded and are now being equipped to the player.
     * @return cosmetics being used by the player
     */
    public Set<Cosmetic> getEquippedCosmetics() {
        throw  new MCNotInstalledException();
    }
}

File: src/main/java/ru/flestiz/ecosmetics/events/PlayerDataLoadEvent.kt
package ru.flestiz.ecosmetics.events

import org.bukkit.event.Event
import org.bukkit.event.HandlerList
import ru.flestiz.ecosmetics.api.Cosmetic
import ru.flestiz.ecosmetics.cache.PlayerData
import ru.flestiz.ecosmetics.utils.MCNotInstalledException

class PlayerDataLoadEvent(
    private val playerData: PlayerData,
    private val equippedCosmetics: MutableSet<Cosmetic>
) : Event(true) {
    init { throw MCNotInstalledException() }

    companion object {
        @JvmStatic fun getHandlerList(): HandlerList = throw MCNotInstalledException()
    }

    override fun getHandlers(): HandlerList = throw MCNotInstalledException()
    fun getPlayerData(): PlayerData = throw MCNotInstalledException()
    fun getEquippedCosmetics(): MutableSet<Cosmetic> = throw MCNotInstalledException()
}